### PR TITLE
[Feature] Mock response status and headers

### PR DESCRIPTION
**Pull request by DRFR0ST**
This PR will allow mocking the response status and headers.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -3,7 +3,19 @@
  * Partial implementation of RequestInit with the addition of "data" property which value will be returned from the mock.
  */
 export type MockOptions = {
+    /** @deprecated use response.data */
```
Comment (DRFR0ST) [inline src/types.ts:6]: I don't feel confidend deprecating the root data property and using the `response.data` instead.
It's logically correct to have the data inside the response object, as it de facto mocks the response body, but it also gets rid of the simplicity of `mock('..', { data: {...} })`.
On the other hand, keeping both seems unnecessary.

**Comment by brianorwhatever** [comment]
Can you please make a release with this PR merged?

**Comment by sadkebab** [comment]
I am waiting for this to be released too

File: src/types.ts (hunk 1 above)
Comment (sadkebab) [reply src/types.ts:6]: I don't think having two ways to do such a same thing is a good idea, I find it unnecessary and confusing. I would suggest deprecating it.

**Comment by brianorwhatever** [comment]
@sadkebab I've added the dist folder to this fork temporarily as I wait for this release https://github.com/brianorwhatever/bun-bagel

**Comment by DRFR0ST** [comment]
Sorry for the delay! Released 🚀

**Pull request merged**